### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ejulio**
Draft while removing dependencies from scrapy

### CODE REVIEW

File: itemloaders/__init__.py [hunk 1]
```diff
@@ -174,48 +151,9 @@ def _process_input_value(self, field_name, value):
...
-
-    def get_css(self, css, *processors, **kw):
-        values = self._get_cssvalues(css, **kw)
-        return self.get_value(values, *processors, **kw)
-
-    def _get_cssvalues(self, csss, **kw):
-        self._check_selector_method()
-        csss = arg_to_iter(csss)
-        return flatten(self.selector.css(css).getall() for css in csss)
+        # if isinstance(self.item, Item):
```
Comment (ejulio) [inline itemloaders/__init__.py:154]: Left commented for now

File: tests/test_base_loader.py [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+            url_in = MapCompose(processor)
+
+        it = dict(name='marta')
+        il = ChildItemLoader(item=it)
+        il.add_value('url', 'text')
+        self.assertEqual(il.get_output_value('url'), ['marta'])
+        il.replace_value('url', 'text2')
+        self.assertEqual(il.get_output_value('url'), ['marta'])
+
+    # def test_add_value_on_unknown_field(self):
```
Comment (ejulio) [inline tests/test_base_loader.py:348]: Not sure if we can test this one for now

File: itemloaders/__init__.py [hunk 3]
```diff
@@ -6,12 +6,9 @@
 from collections import defaultdict
 from contextlib import suppress
 
-from scrapy.item import Item
-from scrapy.loader.common import wrap_loader_context
-from scrapy.loader.processors import Identity
-from scrapy.selector import Selector
```
Comment (kmike) [inline itemloaders/__init__.py:12]: I think most css/xpath code can be brought back by importing Selector from parsel instead of scrapy.selector

**Comment by kmike** [comment]
A meta-question about this PR. https://github.com/scrapy/scrapy/pull/3881 is going to be merged soon (hopefully); it makes quite a serious change to the itemloaders, adding dataclass support. What's the plan of pulling changes to this repo? Should it be done because this PR is merged, or is there a way to do it afterwards (maybe with rebases)?

File: itemloaders/__init__.py (hunk 3 above)
Comment (ejulio) [reply itemloaders/__init__.py:12]: I thought we'd leave `parsel` out for now.
But it makes sense to include ir

**Comment by ejulio** [comment]
@kmike probably, the best approach, would be to merge those changes to `master` in this repo and then merge this PR.
That could help keeping the history sane and avoid some merges

**Comment by kmike** [comment]
hey @ejulio! I think docs may need some files around them, to be buildable.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: Implementation :: CPython',
+        'Programming Language :: Python :: Implementation :: PyPy',
+    ],
+    python_requires='>=3.5',
+    install_requires=[
+        'w3lib==1.21.0',
+        'parsel==1.5.2',
+        'jmespath==0.9.5'
```
Comment (kmike) [inline setup.py:40]: install_requires shouldn't fix package versions; here we can only exclude versions which are known not to work. E.g. w3lib >= 1.21.0 is ok, but w3lib==1.21.0 is not.

Comment (victor-torres) [reply setup.py:40]: We're should also keep a single source of truth. There's a requirements.txt file with the same dependencies.

Comment (kmike) [reply setup.py:40]: A common rule is to use requirements.txt to ping dependencies which are known to work, and install_requires to exclude dependencies which are known not to work. Though I'm not sure why requirements.txt file is needed here.

Comment (ejulio) [reply setup.py:40]: Ok, I'll rename `requirements.txt` to `requirements-dev.txt`

Comment (ejulio) [reply setup.py:40]: Regarding keeping it in two places.
The options I see are:

* `requirements-dev.txt` all dependencies
* `setup.py` package dependencies
* Only `pip install -r requirements-dev.txt`

The options I see are:

* `requirements-dev.txt` only dev dependencies (pytest, flake)
* `setup.py` package dependencies
*  `pip install -r requirements-dev.txt`
* `pip install .`

Is there any other option?

**Comment by ejulio** [comment]
@kmike, regarding the missing docs files, I just copied the text ones.
As probably the ones for building are quite standard, so not sure about keeping the history of those.
But I can add them as well

File: setup.py (hunk 4 above)
Comment (kmike) [reply setup.py]: An approach which we usually follow:

* dependencies which are required to use the package are in install_requires
* dependencies which are required to run tests or other checks (flake8, mypy, sphinx) are in tox.ini file. Sometimes tox.ini can use requirements file (e.g. requirements-tests.txt or tests/requirements.txt), to make it easier to organize. 

We should be running these checks automatically on CI; for local development users shouldn't worry about installing them, they should use tox.

**Comment by kmike** [comment]
@ejulio keeping history is not important for other doc files, but there should be away to build docs (a sphinx project); docs should be built by readthedocs.org eventually.

**Comment by ejulio** [comment]
@kmike I'm working on it in another PR
https://github.com/ejulio/itemloaders/pull/4

File: itemloaders/__init__.py [hunk 5]
```diff
@@ -26,6 +26,69 @@ def unbound_method(method):
...
+
+    .. attribute:: default_output_processor
+
+        The default output processor to use for those fields which don't specify
+        one.
+
+    .. attribute:: default_selector_class
+
+        The class used to construct the :attr:`selector` of this
+        :class:`ItemLoader`, if only a response is given in the ``__init__`` method.
```
Comment (kmike) [inline itemloaders/__init__.py:80]: there is no response support in itemloaders

Comment (kmike) [reply itemloaders/__init__.py:80]: is default_selector_class used in itemloaders, or should it be moved to scrapy.loader?

Comment (ejulio) [reply itemloaders/__init__.py]: Indeed, it's not necessary here.
Removed

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+from setuptools import setup, find_packages
+
+with open('README.md') as f:
+    long_description = f.read()
+
+setup(
+    name='itemloaders',
+    version='0.1',
+    url='https://github.com/ejulio/itemloaders',
```
Comment (kmike) [inline setup.py:9]: could you please change URLs in setup.py?

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,23 @@
+docs/build
```
Comment (kmike) [inline .gitignore:1]: this line is present in .gitignore 2 times

**Comment by kmike** [comment]
Hey @ejulio! The PR looks good to me; I think we can update master to match Scrapy, merge master to this branch (ensuring that changes are preserved), and then merge this branch to master.

**Pull request merged**